### PR TITLE
Rewrite conv activation mkldnn fuse pass tester

### PR DESCRIPTION
**Pull request by HulekJakub**
### PR types
Others

### PR changes
Others

### Describe
Deleted old C++ conv activation mkldnn fuse pass test and updated python one

Currently in PaddlePaddle there are 3 unit-test frameworks - one written in C++ and 2 in Python.
Old fuse pass tests (C++) fuse_pass_tester.cc are checking only if graph has correct number of / type of vertices after fusion. It doesn't check data. So these tests pass even if accuracy drops.
New python framework is superior because it checks both graph fusion and accuracy after fusion.

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py [hunk 1]
```diff
@@ -19,8 +19,8 @@
 from program_config import OpConfig, ProgramConfig, TensorConfig
 
 
-class TestConvActMkldnnFusePass(PassAutoScanTest):
-    r"""
+class TestConvActOneDNNFusePass(PassAutoScanTest):
+    r'''
     x_var   f_var(persistable)
       \       /
         conv2d
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py]: I'd remove this mini-graph

File: python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py [hunk 2]
```diff
@@ -30,38 +30,41 @@ class TestConvActMkldnnFusePass(PassAutoScanTest):
          act
           |
         act_var
-    """
+    '''
 
     def sample_predictor_configs(self, program_config):
         # MKLDNN
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py]:
```suggestion
```

File: python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py [hunk 3]
```diff
@@ -160,34 +165,44 @@ def sample_program_config(self, draw):
...
-            st.sampled_from(["relu", "leaky_relu", "relu6", "swish"])
+            st.sampled_from(
+                [
+                    'relu',
+                    'leaky_relu',
+                    'relu6',
+                    'swish',
+                    'mish',
+                    'hard_sigmoid',
+                    'hard_swish',
```
Comment (Silv3S) [inline python/paddle/fluid/tests/unittests/ir/inference/test_conv_act_onednn_fuse_pass.py]: This list is not complete. Please add all activations from _onednn_reuse.h_ `GetSupportedActivations()` 
```suggestion
                    'abs',
                    'clip',
                    'gelu',
                    'hard_sigmoid',
                    'hard_swish',
                    'leaky_relu',
                    'mish',
                    'relu',
                    'relu6',
                    'sigmoid',
                    'sqrt',
                    'swish',
                    'tanh'
```

**Review by Silv3S — CHANGES_REQUESTED** [review]

File: paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc [hunk 4]
```diff
@@ -26,6 +26,8 @@ using string::PrettyLogDetail;
 
 void ConvActivationMkldnnFusePass::ApplyImpl(Graph* graph) const {
   auto act_types = GetSupportedActivations();
+  act_types.erase(std::remove(act_types.begin(), act_types.end(), "sqrt"),
+                  act_types.end());
```
Comment (Silv3S) [inline paddle/fluid/framework/ir/mkldnn/conv_activation_mkldnn_fuse_pass.cc:30]: Please remove OpCompat related to `sqrt`
- AddOpCompat(OpCompat("sqrt"))
- .EQ("sqrt", 0)

**Review by Silv3S — DISMISSED** [review]

**Review by Silv3S — DISMISSED** [review]

**Review by Silv3S — APPROVED** [review]

**Comment by HulekJakub** [comment]
@XieYunshen please approve changes to unit tests

**Review by XieYunshen — APPROVED** [review]
LGTM for test rewrite

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**